tao_compiler/mlir/disc/transforms/shape_utils.cc

```diff
@@ -449,6 +491,25 @@ LogicalResult ShapeAnalysis::buildBlockDimValueMap(Block* block) {
       }
       auto& decompose = dimValueMulDecompose_[keyDimVal];
       decompose.emplace_back(std::move(decs));
+    } else if (isa<lmhlo::DynamicBroadcastInDimOp>(*op)) {
```

Comment:
is this redundant with line 563?

Reply:
563 is for mhlo dialect. It is lmhlo here.